sys/kern/vm_map.c

```diff
@@ -328,8 +331,17 @@ vm_map_t *vm_map_clone(vm_map_t *map) {
   WITH_MTX_LOCK (&map->mtx) {
     vm_segment_t *it;
     TAILQ_FOREACH (it, &map->entries, link) {
-      vm_object_t *obj = vm_object_clone(it->object);
-      vm_segment_t *seg = vm_segment_alloc(obj, it->start, it->end, it->prot);
+      vm_object_t *obj;
+      vm_segment_t *seg;
+      if (it->flags & VM_SHARED) {
+        refcnt_acquire(&it->object->ref_counter);
+        obj = it->object;
```

Comment:
Do you set `VM_SHARED` anywhere in code?

Reply:
Previously mimiker didn't support sharing the memory. That's why doesn't exist code in kernel that set this particular flag. However now there is a possibility to have shared memory and you can accomplish that by mapping the memory with MAP_SHARED flag and then one of functions for creating vm_segment_t structure will set it. (you can see the example in tests for this functionality)

Reply:
Ok. You call `mmap` with `MAP_SHARED` flag and then you copy it into flags in `sys_mmap` and it works because definitions from `sys/mman.h` fit into `vm_flags_t`.
Please add comments about that assumption.

Reply:
Okay, done

Reply:
I though about adding comment into `vm.h`
> If you do something here then look into `mman.h`
And the same in `mman.h`.

Reply:
Please refer to `sys/kern/mmap.c`. We check correspondence between user and kernel constants there.